core/src/main/scala/com/databricks/labs/remorph/discovery/SnowflakeTableDefinitions.scala

```diff
@@ -0,0 +1,123 @@
+package com.databricks.labs.remorph.discovery
+
+import com.databricks.labs.remorph.parsers.intermediate.{DataType, StructField}
+import com.databricks.labs.remorph.parsers.snowflake.{SnowflakeLexer, SnowflakeParser}
+import org.antlr.v4.runtime.{CharStreams, CommonTokenStream}
+import com.databricks.labs.remorph.parsers.snowflake.SnowflakeTypeBuilder
+
+import java.sql.Connection
+import scala.collection.mutable
+
+class SnowflakeTableDefinitions(conn: Connection) {
+
+  /**
+   * Parses a data type string and returns the corresponding DataType object.
+   *
+   * @param dataTypeString The string representation of the data type.
+   * @return The DataType object corresponding to the input string.
+   */
+  private def getDataType(dataTypeString: String): DataType = {
+    val inputString = CharStreams.fromString(dataTypeString)
+    val lexer = new SnowflakeLexer(inputString)
+    val tokenStream = new CommonTokenStream(lexer)
+    val parser = new SnowflakeParser(tokenStream)
+    val ctx = parser.dataType()
+    val dataTypeBuilder = new SnowflakeTypeBuilder
+    dataTypeBuilder.buildDataType(ctx)
+  }
+
+  private def getTableDefinitionQuery(catalogName: String): String = {
+    s"""WITH column_info AS (
+       |  SELECT
+       |      TABLE_CATALOG,
```

Comment:
```suggestion
       |    TABLE_CATALOG,
```
follow two-space indentation everywhere

Reply:
done